test/06-batch.test.ts

```diff
@@ -0,0 +1,537 @@
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'foo',
+							},
+						},
+						{
+							id: '1',
+							method: 'POST',
+							url: '/university/student',
+							body: {
+								matrix_number: 100004,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'bar',
+							},
+						},
+					],
+				})
+				.expect(
+					400,
+					`"Requests of a batch request must have a method matching one of the following: ${validBatchMethods.join(
+						', ',
+					)}"`,
+				);
+		});
+
+		it('should fail if any of the requests have method GET or DELETE and have a body', async () => {
+			await supertest(testLocalServer)
+				.post('/university/$batch')
+				.send({
+					requests: [
+						{
+							id: '0',
+							method: 'POST',
+							url: '/university/student',
+							body: {
+								matrix_number: 100003,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'foo',
+							},
+						},
+						{
+							id: '1',
+							method: 'GET',
+							url: '/university/student',
+							body: {
+								matrix_number: 100004,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'bar',
+							},
+						},
+					],
+				})
+				.expect(
+					400,
+					'"GET and DELETE requests of a batch request must not have a body"',
+				);
+			await supertest(testLocalServer)
+				.post('/university/$batch')
+				.send({
+					requests: [
+						{
+							id: '0',
+							method: 'POST',
+							url: '/university/student',
+							body: {
+								matrix_number: 100003,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'foo',
+							},
+						},
+						{
+							id: '1',
+							method: 'DELETE',
+							url: '/university/student',
+							body: {
+								matrix_number: 100004,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'bar',
+							},
+						},
+					],
+				})
+				.expect(
+					400,
+					'"GET and DELETE requests of a batch request must not have a body"',
+				);
+		});
+
+		it('should fail if trying to query cross-model in one batch', async () => {
+			await supertest(testLocalServer)
+				.post('/university/$batch')
+				.send({
+					requests: [
+						{
+							id: '0',
+							method: 'POST',
+							url: '/v1/student',
+							body: {
+								matrix_number: 100003,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'foo',
+							},
+						},
+						{
+							id: '1',
+							method: 'POST',
+							url: '/university/student',
+							body: {
+								matrix_number: 100004,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'bar',
+							},
+						},
```

Comment:
For readability it could be a defined validBatchPostRequest object, as it's 5 times the same.


```suggestion
						validBatchPostRequest
```